luarules/gadgets/game_disable_unit_sharing.lua

```diff
@@ -19,15 +19,41 @@ if not gadgetHandler:IsSyncedCode() then
 	return false
 end
 
-if not (Spring.GetModOptions().disable_unit_sharing
-			-- unit market handles the restriction instead if enabled so that selling still works
-			or Spring.GetModOptions().unit_market) then
+local unitSharingMode = Spring.GetModOptions().unit_sharing_mode or "enabled"
+local unitMarketEnabled = Spring.GetModOptions().unit_market
+
+-- If unit sharing is fully enabled and unit market isn't handling restrictions, disable this gadget
+if unitSharingMode == "enabled" and not unitMarketEnabled then
```

Comment:
This is not reasonably equivalent to previous, before it was `NOT (disabled OR market)` yours is `enabled AND NOT market`. I'd expect `disabled OR NOT market`.

Reply:
I really like this suggestion. I had gone in and deleted the crossover code from unit_market in that first PR and the two being all up in each others business is pretty bad. I do like the functional nature of this line, and the fact all the checks are in one place: https://github.com/beyond-all-reason/Beyond-All-Reason/pull/4447/files/1a372178d83de38cbca0bfc28acfceebd97a27be#diff-40c052408aba0b8d61ccb4d5dbd6743bc00d82a1dc532342290f5c0587c7f8c8R46

but I kind of agree with you that the unit market should handle its own logic.

Reply:
what's weird is I tested this. I would expect this not to work.